celest/coordinates/transforms.py

```diff
@@ -0,0 +1,517 @@
+    geodetic altitude above the Earth's surface.
+
+    Parameters
+    ----------
+    itrs : ITRS
+        Itrs coordinates.
+
+    Returns
+    -------
+    np.ndarray
+        1-D array containing the geodetic altitude in kilometres.
+
+    Notes
+    -----
+    This method uses an ellipsoid based model of the Earth to calculate
+    the ellipsoid height in an iterative manner described in "Coordinate
+    Systems in Geodesy" by E. J. Krakiwsky and D.E. Wells. [KW98b]_
+
+    References
+    ----------
+    .. [KW98b] E. J. Krakiwsky and D. E. Wells. Coordinate Systems in
+       Geodesy. Jan. 1998, pp. 31–33.
+    """
+
+    if not isinstance(itrs, ITRS):
+        raise ValueError(f"Input data is in the {itrs.__class__} frame and not"
+                         " the ITRS frame.")
+
+    a, b = WGS84_MAJOR_AXIS_KM, WGS84_MINOR_AXIS_KM
+
+    tol = 10e-10
+
+    itrs_x = itrs.x.to(u.km)
+    itrs_y = itrs.y.to(u.km)
+    itrs_z = itrs.z.to(u.km)
+
+    e = np.sqrt(1 - b ** 2 / a ** 2)
+    p = np.sqrt(itrs_x ** 2 + itrs_y ** 2)
+
+    h = np.sqrt(itrs_x ** 2 + itrs_y ** 2 + itrs_z ** 2) - np.sqrt(a * b)
+    phi = np.arctan((itrs_z / p) * (1 - (e ** 2 * a) / (a + h)) ** -1)
+
+    def new_vals(a, b, e, p, phi, z):
+        N = a / np.sqrt(np.cos(phi) ** 2 + b ** 2 / a ** 2 * np.sin(phi) ** 2)
+        h = p / np.cos(phi) - N
+        phi = np.arctan((z / p) * (1 - (e ** 2 * N) / (N + h)) ** -1)
+
+        return h, phi
+
+    hp, phip = new_vals(a, b, e, p, phi, itrs_z)
+
+    while (np.mean(hp - h) > a * tol) or (np.mean(phip - phi) > tol):
+        h, phi = hp, phip
+        hp, phip = new_vals(a, b, e, p, phi, itrs_z)
+
+    return h
+
+
+def _wgs84_to_itrs(wgs84: WGS84) -> ITRS:
+    """Wgs84 to itrs transformation.
+
+    Parameters
+    ----------
+    wgs84 : WGS84
+        Wgs84 coordinates.
+
+    Returns
+    -------
+    ITRS
+        Itrs coordinates.
+
+    See Also
+    --------
+    _itrs_to_wgs84 : Itrs to wgs84 transformation.
+
+    Notes
+    -----
+    An Earth ellipsoid model is used for the wgs84 to itrs conversion using the
+    methods described in "Coordinate Systems in Geodesy" by E. J. Krakiwsky and
+    D.E. Wells as presented by Christopher Lum. [KW98a]_ [Lum20]_
+
+    References
+    ----------
+    .. [KW98a] E. J. Krakiwsky and D. E. Wells. Coordinate Systems in
+       Geodesy. Jan. 1998.
+    .. [Lum20] Christopher Lum. Geodetic Coordinates: Computing Latitude and
+       Longitude. June 2020.url:https://www.youtube.com/watch?v=4BJ-GpYbZlU.
+    """
+
+    if not isinstance(wgs84, WGS84):
+        raise ValueError(f"Input data is in the {wgs84.__class__} frame and not"
+                         " the WGS84 frame.")
+
+    julian = wgs84.time.to(u.jd2000)
+
+    latitude = wgs84.latitude.to(u.rad)
+    longitude = wgs84.longitude.to(u.rad)
+    height = wgs84.height.to(u.km)
+
+    e = np.sqrt(1 - WGS84_MINOR_AXIS_KM ** 2 / WGS84_MAJOR_AXIS_KM ** 2)
```

Comment:
This will be moved to a constant in the `constants.py` file; call it from there.